contracts/voting-verifier/src/state.rs

```diff
@@ -28,6 +30,19 @@ pub enum Poll {
     ConfirmWorkerSet(WeightedPoll),
 }
 
+impl Poll {
+    pub fn map<F, E>(self, func: F) -> Result<Self, E>
+    where
+        F: FnOnce(WeightedPoll) -> Result<WeightedPoll, E>,
+        E: From<ContractError>,
+    {
+        match self {
+            Poll::Messages(poll) => Ok(Poll::Messages(func(poll)?)),
+            Poll::ConfirmWorkerSet(poll) => Ok(Poll::ConfirmWorkerSet(func(poll)?)),
+        }
+    }
+}
+
```

Comment:
I like the idea of implementing the poll trait for the enum and delegate the calls to the WeightedPoll, the option to invert the control I was considering it before but I didn't liked it because it would introduce some logic into the voting module that seems more fit for the contract itself